elytron/ELY-2078-encryption-for-FilesystemSecurityRealm.adoc

```diff
@@ -0,0 +1,256 @@
+For example the identity with the username `plainUser` will be saved in a file named `OBWGC2LOKVZWK4Q.xml`.
+The identity file being encrypted with a SecretKey that contains a bcrypt hashed password along with 5 attributes would look like the following.
+
+```XML
+<?xml version="1.0" ?>
+<identity xmlns="urn:elytron:1.0">
+    <credentials>
+        <password algorithm="bcrypt" format="enc_base64">RUxZAUMQo2svngST8+WfAg6BoeBtBH0exXBlwAQvN4jCxEuDwPUkniZkgx/3mfetwDJK4rql0jfaigStMAW8mE9iNXG5QQ==</password>
+    </credentials>
+    <attributes>
+        <attribute name="RUxZAUMQUO9rq4PRc3BB61B5oZSepATonjzj1dlUriqbsy//5fw=" value="RUxZAUMQwbcAyojI0gsU0BFxeuBw/zJXR8Umy/jk9tnWrvpRxu0="></attribute>
+        <attribute name="RUxZAUMQ5iZ43cvM0x/RwPxwnqxXZisDmFX0Un+f3pLQz0WDC3o=" value="RUxZAUMQWUAYVdzpdv4sKGaFblcIMsso6x0j5J9oCSHQiDbjJE8="></attribute>
+        <attribute name="RUxZAUMQl1xAhDoJdsMnk7ggw66zyClhUzzB79p859BkQrXazzo=" value="RUxZAUMQCV32L5vunNfLMCISdjyn7myJ0saJdAWreKq1BjnGjfU="></attribute>
+        <attribute name="RUxZAUMQ7t3uOr63lNMwD+Ljsq68ONPwGYdtbVLtdmbY6stZIjg=" value="RUxZAUMQ41IKUmpazIiXd4wQa67I+y35TLtssweZ/1aci7w71ag="></attribute>
+        <attribute name="RUxZAUMQDIjiDvRMr9sOCHnCyh8/uwytBJZOfgCsH0k0itNwcNA=" value="RUxZAUMQi3C21O9+je0b9kXHIsQyoiuSYYafYAFKS3h7iPM9+OE="></attribute>
+    </attributes></identity>
+```
+
+==== Converting Plain Realms to Encrypted Realms
+
+An elytron tool command will be added to let users convert their old unencrypted realms to an realm encrypted with a Secret Key.
+
+The following command would convert a realm located at `standalone/configuration/fs-realm-plain` to an encrypted one at `standalone/configuration/fs-realm-enc` with a Secret Key stored in `mycredstore.cs`.
+
+```bash
+$ ./bin/elytron-tool.sh filesystem-realm-encrypt -i ./standalone/configuration/fs-realm-plain -o ./standalone/configuration/fs-realm-enc -c ./mycredstore.cs
+```
+
+There are 3 mandatory parameters:
+
+- `input-location` - Location of unencrypted realm
+- `output-location` - Location of new encrypted realm
+- `credential-store` - Location of credential store
+
+Along with that there are 7 optional parameters:
+
+- `realm-name` - Name of new encrypted filesystem realm resource. Set to `encrypted-filesystem-realm` by default.
+- `create` - Whether or not the credential store should be dynamically created if it doesn't exist. Set to `true` by default.
+- `secret-key` - The alias of the secret key stored in the credential store file. Set to `key` by default.
+- `hash-encoding` - The hash encoding for the existing filesystem realm. Set to `BASE64` by default.
+- `encoded` - If the original realm has encoded set to true. Set to `true` by default.
+- `levels` - The levels for the existing filesystem realm. Set to `2` by default.
+- `populate` - Whether or not the credential store should be populated with a Secret Key. Set to `true` by default.
+
+
```

Comment:
To cover all the new functionality, `--bulk-convert` parameter should be also mentioned.